ops/ccs-ops-misc/synthetic-data/scripts/synthea-automation/prepare-and-run-synthea.py

```diff
@@ -251,53 +265,40 @@ def check_ranges(properties_file, number_of_benes_to_generate, db_string):
+    overall_validation_result = overall_validation_result and check_single_doesnt_exist('snf_claims', 'fi_doc_clm_cntl_num', fi_num_start, db_string)
+    overall_validation_result = overall_validation_result and check_single_doesnt_exist('hospice_claims', 'fi_doc_clm_cntl_num', fi_num_start, db_string)
+
     ## Since MBI and HICN are incremented in difficult-to-query ways, just check if it exists; if it doesnt exist the range should be fine
     hicn_start = get_props_value(properties_file, "exporter.bfd.hicn_start")
-    query = f"select count(*) from beneficiaries where hicn_unhashed = \'{hicn_start}\'"
-    result = _execute_single_count_query(db_string, query)
-    if result > 0:
-        print(f"(Validation Failure) Start HICN {hicn_start} exists in DB")
-        overall_validation_result = False
-    else:
-        print("(Validation Success) Start HICN does not exist in DB")
+    overall_validation_result = overall_validation_result and check_single_doesnt_exist('beneficiaries', 'hicn_unhashed', hicn_start, db_string)
         
     mbi_start = get_props_value(properties_file, "exporter.bfd.mbi_start")
-    query = f"select count(*) from beneficiaries where mbi_num = \'{mbi_start}\'"
-    result = _execute_single_count_query(db_string, query)
-    if result > 0:
-        print(f"(Validation Failure) Start MBI {mbi_start} exists in DB")
-        overall_validation_result = False
-    else:
-        print("(Validation Success) Start MBI does not exist in DB")
+    overall_validation_result = overall_validation_result and check_single_doesnt_exist('beneficiaries', 'mbi_num', mbi_start, db_string)
         
-    ## Ensure no mbi_hash has more than one bene_id associated with it from any previous loads
-    query = "select count(*) from (select mbi_hash, bene_id, count(bene_id) from beneficiaries_history where mbi_hash IS NOT NULL group by mbi_hash, bene_id having count(bene_id) > 1) as s"
+    ## Ensure no (synthetic) mbi_num has more than one bene_id associated with it from any previous loads
+    ## this takes about 30 seconds in test as of this writing, and 2 seconds in the other envs, but this will increase as more data is added
+    query = 'select count(*) from ('\
+            'select count(*) bene_id_count from ('\
+            'select distinct bene_id, mbi_num '\
+            'from public.beneficiaries '\
+            'where bene_id < 0 and mbi_num IS NOT NULL '\
+            'union '\
+            'select distinct bene_id, mbi_num '\
+            'from public.beneficiaries_history '\
+            'where bene_id < 0 and mbi_num IS NOT NULL '\
+            'union '\
+            'select distinct bene_id, mbi_num '\
+            'from public.medicare_beneficiaryid_history '\
+            'where bene_id < 0 and mbi_num IS NOT NULL '\
+            ') as foo '\
+            'group by mbi_num '\
+            'having count(*) > 1'\
+            ') as s'
```

Comment:
This SQL is a little difficult to read. Would it be possible to format it more clearly?

Reply:
I'm not sure how much more control I have over formatting given its already broken into separate lines; maybe a comment explaining it would help?

Reply:
Added a comment and some indentation to help delineate the layers.